src/Microsoft.VisualStudio.ProjectSystem.Managed/MsBuildUtilities.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    /// <summary>
+    /// Utitlies class to manipulate MsBuild projects.
+    /// </summary>
+    internal abstract class MsBuildUtilities
+    {
+        /// <summary>
+        /// Gets a project property.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="propertyName">Name of the property.</param>
+        /// <returns>Requested project property. Null if the property is not present.</returns>
+        public static ProjectPropertyElement GetProperty(ProjectRootElement project, string propertyName)
+        {
+            Requires.NotNull(project, "project");
+
+            return project.Properties
+                .Where(p => string.Equals(p.Name, propertyName, StringComparison.OrdinalIgnoreCase))
+                .FirstOrDefault();
```

Comment:
The dimensions that we deal here are Platform and Configuration, which are usually not usually conditioned. But modifying a Msbuild Construction model does not ensure correctness. These are one-time operation or user-triggered operations, for which we can take a perf hit.

To ensure correctness you should be using [this](http://index/?leftProject=Microsoft.Build&leftSymbol=rrmvongpczat&file=Definition%5CProjectProperty.cs&rightSymbol=rz6kyqzah8cm) instead. This gives the XML element of the project property which is actually part of the Msbuild Evaluation Model rather than the Construction model. If we do this, then we dont have to worry whether the property is conditioned or not. It just works